test/framework/kogitoapp.go

```diff
@@ -34,144 +30,92 @@ import (
 )
 
 const (
-	boxExamplesPath       = "../../deploy/examples"
-	mavenArgsAppendEnvVar = "MAVEN_ARGS_APPEND"
-	mavenMirrorURLEnvVar  = "MAVEN_MIRROR_URL"
+	boxExamplesPath = "../../deploy/examples"
 )
 
-// KogitoAppDeployment defines the elements for the deployment of a kogito application
-type KogitoAppDeployment struct {
-	AppName            string
-	GitSourceURI       string
-	GitSourceReference string
-	ContextDir         string
-	Runtime            v1alpha1.RuntimeType
-	Native             bool
-	Persistence        bool
-	Events             bool
-	Labels             map[string]string
-}
-
 // DeployService deploy a Kogito service
-func DeployService(namespace string, installerType InstallerType, kogitoAppDeployment KogitoAppDeployment) error {
-	GetLogger(namespace).Infof("%s deploy %s example %s with name %s, native %v, persistence %v, events %v and labels %v", installerType, kogitoAppDeployment.Runtime, kogitoAppDeployment.ContextDir, kogitoAppDeployment.AppName, kogitoAppDeployment.Native, kogitoAppDeployment.Persistence, kogitoAppDeployment.Events, kogitoAppDeployment.Labels)
+func DeployService(namespace string, installerType InstallerType, kogitoApp *v1alpha1.KogitoApp) error {
+	GetLogger(namespace).Infof("%s deploy %s example %s with name %s, native %v, persistence %v, events %v and labels %v", installerType, kogitoApp.Spec.Runtime, kogitoApp.Spec.Build.GitSource.ContextDir, kogitoApp.Name, kogitoApp.Spec.Build.Native, kogitoApp.Spec.EnablePersistence, kogitoApp.Spec.EnableEvents, kogitoApp.Spec.Service.Labels)
+
+	setupBuildImageStreams(kogitoApp)
```

Comment:
should be called in `crDeployService` or in `cliDeployService` you should check against this kogitoApp and not the config